src/dataset-builder/ConceptSearch.ts

```diff
@@ -0,0 +1,130 @@
+import { IconId } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import { Fragment, useEffect, useState } from 'react';
+import { div, h, h2 } from 'react-hyperscript-helpers';
+import { ActionBar } from 'src/components/ActionBar';
+import { Link, spinnerOverlay } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { SimpleTable } from 'src/components/table';
+import { StringInput } from 'src/data-catalog/create-dataset/CreateDatasetInputs';
+import { GetConceptsResponse } from 'src/dataset-builder/DatasetBuilderUtils';
+import { SnapshotBuilderConcept as Concept, SnapshotBuilderDomainOption } from 'src/libs/ajax/DataRepo';
+import { DataRepo } from 'src/libs/ajax/DataRepo';
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData';
+import colors from 'src/libs/colors';
+
+import { PAGE_PADDING_HEIGHT, PAGE_PADDING_WIDTH } from './constants';
+
+type ConceptSearchProps = {
+  readonly initialSearch: string;
+  readonly domainOption: SnapshotBuilderDomainOption;
+  readonly onCancel: () => void;
+  readonly onCommit: (selected: Concept[]) => void;
+  readonly onOpenHierarchy: (
+    domainOption: SnapshotBuilderDomainOption,
+    selected: Concept[],
+    searchText: string
+  ) => void;
+  readonly actionText: string;
+  readonly datasetId: string;
+  readonly initialCart: Concept[];
+};
+
+export const ConceptSearch = (props: ConceptSearchProps) => {
+  const { initialSearch, domainOption, onCancel, onCommit, onOpenHierarchy, actionText, datasetId, initialCart } =
+    props;
+  const [search, setSearch] = useState<string>(initialSearch);
+  const [cart, setCart] = useState<Concept[]>(initialCart);
+  const [concepts, searchConcepts] = useLoadedData<GetConceptsResponse>();
+  useEffect(() => {
+    void searchConcepts(() => {
+      return DataRepo().dataset(datasetId).searchConcepts(domainOption.root, search);
+    });
+  }, [search, datasetId, domainOption.root, searchConcepts]);
+  return h(Fragment, [
+    div({ style: { padding: `${PAGE_PADDING_HEIGHT}rem ${PAGE_PADDING_WIDTH}rem` } }, [
+      h2({ style: { display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            onClick: onCancel,
+            'aria-label': 'cancel',
+          },
+          [icon('left-circle-filled', { size: 32 })]
+        ),
+        div({ style: { marginLeft: 15 } }, [domainOption.category]),
+      ]),
+      h(StringInput, {
```

Comment:
Would we rather use `ValidatedInput` rather than `StringInput`? I think StringInput is localized to the Create-Dataset directory in the catalog. Another option could be whether we should think about lifting some of those inputs to a higher level? What do you think? I think we have a philosophy to avoid sharing code across different feature areas where possible, and may want to lift this component up if we are going to reuse it. (This may already be used in multiple places, in which case it goes extra to think about lifting it up)

Reply:
I've changed this to use `DelayedSearchInput` which is from `components`, so it no longer depends on the catalog component type. It does seem like we have a lot of different text input types in the codebase and it'd be nice to use common types where possible. Maybe this is something the storybook docs could help with.